redash/models/__init__.py

```diff
@@ -130,6 +133,17 @@ def to_dict(self):
         }
 
 
+@python_2_unicode_compatible
+class TableMetadataQueriesLink(db.Model):
+    table_id = Column(db.Integer, db.ForeignKey("table_metadata.id", ondelete="CASCADE"), primary_key=True)
+    query_id = Column(db.Integer, db.ForeignKey("queries.id", ondelete="CASCADE"), primary_key=True)
```

Comment:
Are we sure we want to cascade delete in both cases?

Reply:
I think so. If a table's metadata is deleted, we shouldn't keep its association with a query. And if a query is deleted, we shouldn't keep its association with a table's metadata.

Is there a case you felt it shoudn't cascade?

Reply:
I'm worried users may not know about associations and delete a query without understanding what this results in. I can't come up with a good alternative though. Let's ship and reconsider later..